builder/optimizations.cc

```diff
@@ -797,35 +797,38 @@ class in_place_aliaser : public stmt_mutator {
     }
   }
 
+  var alloc_of_buffer(var i) const {
+    std::optional<var> alloc = aliases.lookup(i);
+    return alloc ? *alloc : i;
+  }
+
   void visit(const call_stmt* op) override {
     set_result(op);
 
-    std::size_t output = 0;
-    for (var i : op->inputs) {
-      std::optional<var> i_root = aliases.lookup(i);
-      if (!i_root) {
-        // This was an input to the pipeline, we can't alias it.
-        continue;
-      }
+    for (std::size_t o = 0; o < op->outputs.size(); ++o) {
+      for (std::size_t i = 0; i < op->inputs.size(); ++i) {
+        if ((op->attrs.allow_in_place & (1 << (o * op->inputs.size() + i))) == 0) {
```

Comment:
In the unlikely event that we have more inputs and outputs than there are in the int, this will produce UB, I think.

Reply:
Added an assert. Also see comment in header.